samples/cli/lib/samples/impersonate.rb

```diff
@@ -0,0 +1,61 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'google/apis/gmail_v1'
+
+module Samples
+  # Examples for the Gmail API
+  #
+  # Sample usage:
+  #
+  #     $ ./google-api-samples impersonate update_email_signature 'My new signature!' \
+  #       --impersonated_email='user@example.com'
+  #
+  class Gmail < BaseCli
+    Gmail = Google::Apis::GmailV1
+
+    desc 'impersonate and update email signature', 'Update the email signature of another user'
+
+    method_option :impersonated_email, type: :string, required: true
+
+    def update_email_signature(new_signature_content)
```

Comment:
This seems to use the same API as the existing sample, could you move this function to that class?